src/View/Helper/FormHelper.php

```diff
@@ -1502,13 +1502,21 @@ protected function _magicOptions($fieldName, $options, $allowOverride)
             unset($options['step']);
         }
 
+        $validationLength = $context->getMaxLength($fieldName);
```

Comment:
Shouldn't you first check if the field type is one of those in `$allowedTypes` below? If it's not then this method call is unnecessary overhead for other input types (e.g. `select`) which don't need `maxlength` attribute.

So I think this additional code should be move inside the `if` after `$allowedTypes` is initialized.

Reply:
I understand that maxlength won't be set for e.g. for a textarea. But the `$context->getMaxLength()` call will still be made, which is unnecessary. That's my issue.

Reply:
I think we have to rewrite this. Please check out my new file.
This will checking maxlength param too. Is this wrong? What you recommend?